omc3/kmod_averages.py

```diff
@@ -0,0 +1,197 @@
+import tfs
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+
+from pathlib import Path
+from generic_parser import DotDict
+from generic_parser.entrypoint_parser import EntryPointParameters, entrypoint
+from omc3.utils import logging_tools
+from omc3.utils.iotools import PathOrStr
+
+LOG = logging_tools.get_logger(__name__)
+
+
+def kmod_average_params():
+    """
+    A function to create and return EntryPointParameters for Kmod average.
+    """
+    params = EntryPointParameters()
+    params.add_parameter(name="meas_paths",
+                         required=True,
+                         nargs='+',
+                         help="Directories of Kmod results to import")
+    params.add_parameter(name="ip",
+                         required=True,
+                         type=int,
+                         help="Specific ip to average over")
+    params.add_parameter(name="beta",
+                         required=True,
+                         type=float,
+                         help="Model beta of measurements")
+    params.add_parameter(name="output_dir",
+                         required=True,
+                         type=PathOrStr,
+                         help="Path to the directory where to write the output files.")
+    return params
+
+
+@entrypoint(kmod_average_params(), strict=True)
+def average_kmod_results_entrypoint(opt: DotDict) -> None:
+    opt.meas_paths = [Path(m) for m in opt.meas_paths]
+    opt.output_dir = Path(opt.output_dir)
+    opt.output_dir.mkdir(exist_ok=True)
+    averaged_results = get_average_betastar_results(opt)
+    averaged_bpm_results = get_average_bpm_betas_results(opt)
+    plot_results(opt, averaged_results)
+
+
+def get_average_betastar_results(opt):
+    """
+    Calculate the average betastar results for the given parameters.
+
+    Args:
+        opt: The parameters for calculation.
+
+    Returns:
+        The final results as a DataFrame.
+    """
+    final_results = []
+    for beam in [1, 2]:
+        all_dfs = []
+        for mpath in opt.meas_paths:
+            print(mpath)
+            all_dfs.append(tfs.read(mpath / f'B{beam}' / 'results.tfs').drop(columns=['LABEL', 'TIME']))
+
+        panel = np.array(all_dfs)
+        # Calculate mean and std along the new axis (axis=0)
+        mean_df = pd.DataFrame(panel.mean(axis=0), index=all_dfs[0].index, columns=all_dfs[0].columns)
+        std_df = pd.DataFrame(panel.std(axis=0), index=all_dfs[0].index, columns=all_dfs[0].columns)
+
+        for column in mean_df.columns:
+            if not column.startswith('ERR'):
+                mean_df[f'ERR{column}'] = std_df[column]
```

Comment:
@fscarlier  you replace the present ERR-Columns with the standard deviation over the measurements,
ignoring the ERR-columns from the K-mod file. Shouldn't these errors be taken into account? I am sure we do that in other places when averaging over data with errors.